### DISCUSSION TITLE
Renaming a car?

### DISCUSSION THREAD

**Discussion by Kosta-Github** [teslamate-org/teslamate#1938] [Q&A] [answered]
Is there a way to easily rename a car? SQL statement would be fine, too; but I donˋt know the DB schema…

Background: I sold my old M3 and received my new MY and want to keep the history of the old one, but the new one should be the first one on all the dashboards now. Unfortunately the new one has a name alphabetically after the old one…

**Comment by DrMichael** [comment]
You might find https://github.com/adriankumpf/teslamate/discussions/1224 helpful...

**Comment by leewillis77** [comment]
Also, please also vote on #1751 and/or if you're technical any comments on #1904 would be welcome.

**Comment by Kosta-Github** [comment, marked as answer]
Just for closing the loop here: the SQL commands for renaming a car would be:

1. list all cars in the DB:
   ```
   docker-compose exec -T database psql -U teslamate << .                            
       SELECT * FROM cars;
   .
   ```
2. renaming one car from `old_car_name` to `new_car_name`:
   ```
   docker-compose exec -T database psql -U teslamate << .                            
       UPDATE cars SET name='new_car_name' WHERE name='old_car_name';
   .
   ```

Note, that `database` refers to the name of the `DB` service in your `docker-compose` config file; if you renamed this service, e.g., to `teslamate-db`, you would also need to use the corresponding service name here: `docker-compose exec -T teslamate-db ...`

**Reply by horst789github** [reply]
dont work for me:

with root: service "teslamate-db" is not running container #1
with user: no configuration file provided: not found

**Comment by leewillis77** [comment]
I think you had `old_car_name` and `new_car_name` the wrong way around in your example, so should be...

```
docker-compose exec -T database psql -U teslamate << .                            
    UPDATE cars SET name='new_car_name' WHERE name='old_car_name';
.
```

**Comment by Kosta-Github** [comment]
Good catch, fixed it above as well…

**Comment by bbrendon** [comment]
I think car renaming is auto-magically handled now? My car was renamed any I don't see the old name in `SELECT * FROM cars;`
